freetextresponse/mixins.py

```diff
@@ -0,0 +1,32 @@
+"""
+Mixins for the Free Text Response XBlock
+"""
+import datetime
+import pytz
+from django.conf import settings
+
+
+class EnforceDueDates(object):
```

Comment:
Pylint is breaking here.
It's set to require 2 public methods per class.

I basically had added that `before_due()` function so it would have 2 public methods

Reply:
@caesar2164 Can you paste the actual error message?

Reply:
```
R:  9, 0: Too few public methods (1/2) (too-few-public-methods)
```

Reply:
I could add a second method?
```
   def has_graceperiod(self):
        """
        Determine if course has a graceperiod set
        """
        if self.graceperiod is not None:
            return True
        return False
```

Reply:
No, we should never write unused code, just to satisfy static analysis tools.

People commonly miss the _spirit_ of the rule,
in favor of the _letter_ of the law.

Pylint errors don't mean:

> This is never acceptable code; change it immediately!

It's more nuanced; what they're really saying is:

> While syntactically valid, this design pattern is _often_ indicative of a mistake/oversight and/or can _often_ lead to issues later on.

Specifically here, what they're saying is:

> Generally speaking, classes should have more than one method. If your implementation doesn't, then maybe a dedicated `class` is overkill and a lighter abstraction is more appropriate.

In this case, our response is along the lines of:

> Yeah, we know it's lightweight. It's actually not meant to be a standalone class, but rather mixed-in to base class. We appreciate your concern, but it's nothing to worry about.

And we "respond" to that by ignoring the pylint error.

Similarly, in an ideal world, ignoring pylint doesn't mean:

> I'm too lazy to fix this

Instead it _should_ mean:

> WONTFIX: for reasons